### ISSUE TITLE
PE2 Test6_1

### ISSUE THREAD

**Issue by e0531336** [nus-cs2030/2021-s1#325]
Hi, anyone know where I can get the .jsh file for test6_1? Having issues trying to fix my 92/100 to 100/100 unless............

**Comment by zimmmz** [comment]
Hi, err I don't think we will have access to that. Why not you share your error from cc here so that people that encounter the same issue might be able to help?

**Comment by hhhhanz** [comment]
6_1 is a hidden test case so we wont have access

**Comment by e0531336** [comment]
![image](https://user-images.githubusercontent.com/72070516/98976678-9d356400-2552-11eb-8918-08120bf060b3.png)
and 
![image](https://user-images.githubusercontent.com/72070516/98976720-a9212600-2552-11eb-945f-65705e4bceac.png)

**Comment by zimmmz** [comment]
@e0531336 for the first ss i think is because you forget to remove your sword in the previous room when the sword was taken over to the next room.  For the second ss I am not very sure but it seems like your output is like one step ahead means it was tick()?

**Comment by jiayi-lim** [comment]
For the second ss, if there is no previous room the method should return this without tick()!

**Comment by jerimiglo** [comment]
yup!!! i returned this without tick() if there is no room and it worked for me!

**Comment by adelineT-jy** [comment]
For the second ss. I have the same problem but i resolved it. 
What happens here is that you can back() all the way to the first room and you can back() again in the first room. By doing back() in the first room, you should get back the _same output_ as when you ask for that same room.

Hope this clarifies your question

**Comment by roopaniedu** [comment]
Hey i had similar problems for test case 6_1 and managed to figure out the hidden test cases.
`Room i = new Room("a").go(x -> new Room("b").add(new Sword()).tick(takeSword)).go(x -> new Room("c").add(new Troll())).go(x -> new Room("d"))`
`Room n = i.tick(dropSword).back()`
`Room o = n.back()`
`Room p = o.back()`

**Comment by zhiqitan** [comment]
hi, was facing similar problems also. How can I handle the the first error ar? or anyone know the test case for the first one ?
<img width="1190" alt="Screenshot 2020-11-14 at 1 13 36 AM" src="https://user-images.githubusercontent.com/67494720/99101783-08953980-2618-11eb-9f67-92c339a3ee03.png">

**Comment by jiayi-lim** [comment]
@zhiqitan The first error occurred because you added the sword to the wrong place! When u `go() `to a new room and brought a sword with you, the sword will be added to the front of the list. Whereas when u `back()` and brought a sword with you, the sword will be added to the back of the list.

**Comment by aucjl** [comment]
hello! i faced similar issues as @zhiqitan for the first one and followed as @jiayi-lim advised but still not too sure why im still getting the same error. :( 
![image](https://user-images.githubusercontent.com/69624115/99154216-09e26700-26e9-11eb-9d91-493242f19b5d.png)

**Comment by zhiqitan** [comment]
@aucjl hi, jiayi-lim advice works for me. you can try to debug with this test case, and the test code for the problem is 
Room j = i.back(); 
not sure which part goes wrong, but you may think of it like a game, so whenever where you go or back, if you taken a sword, the sword will be added to the back of your items list, so maybe you can check back your code. One thing you can check is your previous room has the sword alrd, so you need to remove the old sword when you back to the previous room as the sword by right has been brought along with you to other room and should have no longer in the previous room.

> Hey i had similar problems for test case 6_1 and managed to figure out the hidden test cases.
> `Room i = new Room("a").go(x -> new Room("b").add(new Sword()).tick(takeSword)).go(x -> new Room("c").add(new Troll())).go(x -> new Room("d"))`
> `Room n = i.tick(dropSword).back()`
> `Room o = n.back()`
> `Room p = o.back()`

**Comment by KaiyueSong** [comment]
> Hey i had similar problems for test case 6_1 and managed to figure out the hidden test cases.
> `Room i = new Room("a").go(x -> new Room("b").add(new Sword()).tick(takeSword)).go(x -> new Room("c").add(new Troll())).go(x -> new Room("d"))`
> `Room n = i.tick(dropSword).back()`
> `Room o = n.back()`
> `Room p = o.back()`

Hi, I used this locally to test my code, and there is no problem..
But I still have an extra "Sword is slimmering" when I upload my code to cc....
Any hint why??

**Comment by e0531336** [comment]
I know its a bit late, but this was helpful for me!

/open Things.java
/open Candle.java
/open Room.java
/open Troll.java
/open Sword.java
/open actions.jsh

Room d = new Room("a").go(x -> new Room("b").add(new Sword()).tick(takeSword)).go(x -> new Room("c").add(new Troll())).go(x -> new Room("d"))
Room dd = d.tick(dropSword)
Room c = dd.back()
Room b = c.back()
Room a = b.back()

d
dd
c
b
a
a
